### PR TITLE
fix:Dashboard stats changes for Simplified report closure

### PR DESCRIPTION
**Pull request by arjunaj5**
https://app.clickup.com/t/85zrpcyvg
Slack thread: https://fylein.slack.com/archives/C03JJJP4MSL/p1677559257845899

[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=iTffk3Y)

Changes in this PR:

1. Service function to check for nonreimbursable accounts. (accounts having CCC only or CCC + paid by company payment modes)
2. Dashboard stats changes to remove Approved and Payment spending stats for non-reimbursable only accounts in the new flow
3. Dashboard stats changes to change `payment pending` stat to `processing` stat for reimbursable accounts in the new flow.

### CODE REVIEW

**Review by gokulk16 — APPROVED** [review]

File: src/app/core/models/org_user_settings.model.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { AccountType } from '../enums/account-type.enum';
+import { AllowedPaymentModes } from './allowed-payment-modes.enum';
```
Comment (yksolanki9) [inline src/app/core/models/org_user_settings.model.ts:2]: Is the import needed?

File: src/app/fyle/dashboard/stats/stats.component.html [hunk 2]
```diff
@@ -55,16 +55,35 @@
...
+                [currencySymbol]="currencySymbol$ | async"
+                [name]="'Payment Pending'"
+                [reportState]="ReportStates.PAYMENT_PENDING"
+                [count]="(paymentPendingStats$ | async)?.count"
+                [value]="(paymentPendingStats$ | async)?.sum"
+                [loading]="reportStatsLoading"
+                (badgeClicked)="goToReportsPage($event)"
+              ></app-stat-badge>
+            </ng-container>
+            <ng-container *ngIf="simplifyReportsSettings.enabled">
```
Comment (yksolanki9) [inline src/app/fyle/dashboard/stats/stats.component.html:71]: Put this inside `ng-template`

File: src/app/fyle/dashboard/stats/stats.component.html [hunk 3]
```diff
@@ -55,16 +55,35 @@
...
-            [currencySymbol]="currencySymbol$ | async"
-            [name]="'Payment Pending'"
-            [reportState]="ReportStates.PAYMENT_PENDING"
-            [count]="(paymentPendingStats$ | async)?.count"
-            [value]="(paymentPendingStats$ | async)?.sum"
-            [loading]="reportStatsLoading"
-            (badgeClicked)="goToReportsPage($event)"
-          ></app-stat-badge>
+          <ng-container *ngIf="simplifyReportsSettings$ | async as simplifyReportsSettings; else loading">
+            <ng-container *ngIf="!simplifyReportsSettings.enabled">
```
Comment (yksolanki9) [inline src/app/fyle/dashboard/stats/stats.component.html:59]: Use else if `<ng-container *ngIf="...; else "> `

File: src/app/fyle/dashboard/stats/stats.component.ts [hunk 4]
```diff
@@ -6,7 +6,7 @@ import { delay, finalize, map, switchMap, tap } from 'rxjs/operators';
 import { CurrencyService } from '../../../core/services/currency.service';
 import { Params, Router } from '@angular/router';
 import { NetworkService } from '../../../core/services/network.service';
-import { concat, of, Subject } from 'rxjs';
+import { concat, interval, of, Subject } from 'rxjs';
```
Comment (yksolanki9) [inline src/app/fyle/dashboard/stats/stats.component.ts:9]: Remove unused imports

File: src/app/fyle/dashboard/stats/stats.component.ts [hunk 5]
```diff
@@ -148,6 +156,7 @@ export class StatsComponent implements OnInit {
     that.initializeReportStats();
     that.initializeExpensesStats();
     that.orgSettingsService.get().subscribe((orgSettings) => {
+      this.isNonReimbursableOrg = this.paymentModeService.isNonReimbursableOrg(orgSettings.payment_mode_settings);
```
Comment (yksolanki9) [inline src/app/fyle/dashboard/stats/stats.component.ts:159]: Convert this to observable, else user may see old value

**Review by yksolanki9 — CHANGES_REQUESTED** [review]

File: src/app/core/models/org_user_settings.model.ts (hunk 1 above)
Comment (arjunaj5) [reply src/app/core/models/org_user_settings.model.ts]: no, removed it

File: src/app/fyle/dashboard/stats/stats.component.html (hunk 2 above)
Comment (arjunaj5) [reply src/app/fyle/dashboard/stats/stats.component.html]: done

File: src/app/fyle/dashboard/stats/stats.component.html (hunk 3 above)
Comment (arjunaj5) [reply src/app/fyle/dashboard/stats/stats.component.html]: yes, done 🚀

File: src/app/fyle/dashboard/stats/stats.component.ts (hunk 4 above)
Comment (arjunaj5) [reply src/app/fyle/dashboard/stats/stats.component.ts]: :done

File: src/app/fyle/dashboard/stats/stats.component.ts (hunk 5 above)
Comment (arjunaj5) [reply src/app/fyle/dashboard/stats/stats.component.ts]: converted to observable.

**Review by yksolanki9 — APPROVED** [review]

**Pull request merged**